tools/compile_tf_graph.py

```diff
@@ -501,6 +610,7 @@ def __init__(self, beam_size,
     source = self.sources[0]
     assert source.output.is_batch_major and len(source.output.shape) == 1
     scores_in = source.output.placeholder
+    # TODO infer input_type from source layer class 'softmax' or 'linear' with 'log_softmax' activation
```

Comment:
What do you mean? There is nothing to be inferred. This is configured.

Reply:
you mean input_type is 'prob' or 'log_prob' is automatically detected when creating this ChoiceStateVarLayer ? I didn't see this

Reply:
No, input_type is something which is configured. This is not inferred. And this is by design.